### PR TITLE
Chore/upgrade openzeppelin contracts

### PR DESCRIPTION
**Pull request by AndriianChestnykh**

### CODE REVIEW

File: contracts/identitytreestore/IdentityTreeStore.sol [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // SPDX-License-Identifier: GPL-3.0
-pragma solidity 0.8.16;
+pragma solidity 0.8.20;
```
Comment (vmidyllic) [inline contracts/identitytreestore/IdentityTreeStore.sol:2]: can we update pragma for already deploy contract? will it have influence on upgrade? (for each contract)

File: contracts/state/State.sol [hunk 2]
```diff
@@ -70,7 +70,7 @@ contract State is Ownable2StepUpgradeable, IState {
         verifier = verifierContractAddr;
         _setDefaultIdType(defaultIdType);
         _gistData.initialize(MAX_SMT_DEPTH);
-        __Ownable_init();
+        __Ownable_init(msg.sender);
```
Comment (vmidyllic) [inline contracts/state/State.sol:73]: looks like a change in State contract, version must be increased?

File: contracts/verifiers/ZKPVerifier.sol [hunk 3]
```diff
@@ -37,6 +37,8 @@ contract ZKPVerifier is IZKPVerifier, Ownable {
     // (see https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#storage-gaps)
     uint256[47] __gap_after;
 
+    constructor() Ownable(msg.sender) {}
```
Comment (vmidyllic) [inline contracts/verifiers/ZKPVerifier.sol:40]: is it needed? was it a bug?

File: package.json [hunk 4]
```diff
@@ -51,8 +51,8 @@
   },
   "dependencies": {
     "@nomiclabs/hardhat-etherscan": "^3.0.3",
-    "@openzeppelin/contracts": "^4.7.3",
-    "@openzeppelin/contracts-upgradeable": "^4.8.2",
+    "@openzeppelin/contracts": "^5.0.1",
```
Comment (vmidyllic) [inline package.json:54]: how do we keep old version of contract?

File: contracts/identitytreestore/IdentityTreeStore.sol (hunk 1 above)
Comment (AndriianChestnykh) [reply contracts/identitytreestore/IdentityTreeStore.sol:2]: It shouldn't. The version is minor upgrade so no breaking changes and different behaviour in logic expected. Most important, storage layout will not be changed for sure.

File: contracts/state/State.sol (hunk 2 above)
Comment (AndriianChestnykh) [reply contracts/state/State.sol]: I've rolled this change back cuz added two OpenZeppelin versions to the repo.

File: package.json (hunk 4 above)
Comment (AndriianChestnykh) [reply package.json:54]: My bad. I've put you in review before the PR is finalised.
Now I've added two versions of OpenZeppelin.

**Review by vmidyllic — APPROVED** [review]

**Pull request merged**